diff -ru whack_src_1_0_0/source/java/org/xmpp/packet/JID.java whack_src_fixed/source/java/org/xmpp/packet/JID.java
--- whack_src_1_0_0/source/java/org/xmpp/packet/JID.java	2008-07-15 06:51:09.000000000 +1000
+++ whack_src_fixed/source/java/org/xmpp/packet/JID.java	2009-07-07 20:13:27.451727000 +1000
@@ -20,9 +20,9 @@
 
 package org.xmpp.packet;
 
-import org.jivesoftware.stringprep.IDNA;
-import org.jivesoftware.stringprep.Stringprep;
-import org.jivesoftware.stringprep.StringprepException;
+import gnu.inet.encoding.IDNA;
+import gnu.inet.encoding.Stringprep;
+import gnu.inet.encoding.StringprepException;
 
 import java.io.Serializable;
 import java.util.Collections;
@@ -348,7 +348,7 @@
         }
         // Stringprep (node prep, resourceprep, etc).
         try {
-            if (!stringprepCache.containsKey(node)) {
+            if (node != null && !stringprepCache.containsKey(node)) {
                 this.node = Stringprep.nodeprep(node);
                 // Validate field is not greater than 1023 bytes. UTF-8 characters use two bytes.
                 if (this.node != null && this.node.length()*2 > 1023) {
@@ -376,7 +376,11 @@
             else {
                 this.domain = domain;
             }
-            this.resource = resourceprep(resource);
+            if (resource == null) {
+              this.resource = null;
+            } else {
+              this.resource = resourceprep(resource);
+            }
             // Validate field is not greater than 1023 bytes. UTF-8 characters use two bytes.
             if (resource != null && resource.length()*2 > 1023) {
                 throw new IllegalArgumentException("Resource cannot be larger than 1023 bytes. " +
